File: backend/spring-boot/src/main/java/org/bugzkit/api/auth/jwt/service/AccessTokenService.java
package org.bugzkit.api.auth.jwt.service;

import java.util.Set;
import org.bugzkit.api.user.payload.dto.RoleDTO;

public interface AccessTokenService {
  String create(Long userId, Set<RoleDTO> roleDTOs);

  void check(String token);

  void invalidate(String token);

  void invalidateAllByUserId(Long userId);
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/auth/jwt/service/AccessTokenService.kt
package org.bootstrapbugz.api.auth.jwt.service

import org.bootstrapbugz.api.user.payload.dto.RoleDTO

interface AccessTokenService {
  fun create(userId: Long?, roleDTOs: Set<RoleDTO>): String

  fun check(token: String)

  fun invalidate(token: String)

  fun invalidateAllByUserId(userId: Long?)
}
